### PR TITLE
st_archive interactive resubmit fix

### LINKED ISSUE #2547: Running case.st_archive with RESUBMIT > 0 submits case.run jobs
One of our users complained about this, and it seems very unintuitive from a UI perspective for people who run case.st_archive from the login node.
My thought on a solution to this is to add an argument to case.st_archive which enables resubmitting, and have case.submit set that option for the archiving jobs.

### PR DESCRIPTION
**Pull request by mfdeakin-sandia**
This detects whether the short term archiver is being run from the queue or not, and if not, does not perform any resubmits.
This still needs a unit test, but that should be easy

Test suite: scripts_regression_tests.py
Test status: in progress

Fixes #2547 

User interface changes?: N

Code review: @jedwards4b @jgfouca

### CODE REVIEW

**Comment by mfdeakin-sandia** [comment]
I got a failure from `PRE.f19_f19.ADESP.sandiatoss3_intel.fake_testing_only` in `Z_FullSystemTest` that looks like restart files were moved before they should have been? Can I ignore that? All the others in the full system test passed

File: config/xml_schemas/config_batch.xsd [hunk 1]
```diff
@@ -78,6 +79,9 @@
   <!-- walltime_format: time format expected by batch system for the wall clock time field -->
         <xs:element minOccurs="0" ref="walltime_format"/>
 
+  <!-- batch_jobid_env_var: environment variable set by the batch system indicating the
+       job id for the submitted job -->
+        <xs:element minOccurs="0" ref="batch_jobid_env_var"/>
```
Comment (jedwards4b) [inline config/xml_schemas/config_batch.xsd]: you have minOccurs = 0 but it looks like this is now a required field?

Comment (mfdeakin-sandia) [reply config/xml_schemas/config_batch.xsd]: I check and verify it exists in `EnvBatch.is_submission_job`, so it shouldn't be not required

Comment (jedwards4b) [reply config/xml_schemas/config_batch.xsd]: okay - what about the case in which you are not using a batch system at all?

Comment (mfdeakin-sandia) [reply config/xml_schemas/config_batch.xsd]: TBC, I do prefer the idea of adding a command line argument to case.st_archive; no new xml and doesn't have this issue

Comment (mfdeakin-sandia) [reply config/xml_schemas/config_batch.xsd]: @jedwards4b Another reason to prefer the command line option - it solves the issue of RESUBMIT being set by the user during a run in the other PR I have. If it's okay, I'll just add that method to that branch, and close this PR?

**Pull request closed without merging**